### ISSUE TITLE
Updating with no data does not remove tooltip handling

### ISSUE THREAD

**Issue by jgrund** [novus/nvd3#998]
When a chart is updated with an empty data set, Tooltip handlers are no cleared. 

When a user mouses over a chart, they can still see the tooltip from the previous data set.

**Comment by Resseguie** [comment]
I've had the same problem.

**Comment by ilariuss** [comment]
I have this problem at this moment, but I don´t know what to do. Thanks

**Comment by wayoutmind** [comment]
+1

**Comment by daumann** [comment]
+1

**Comment by liquidpele** [comment]
Marked as a bug... will look into it when I get a chance but if anyone else wants to take a stab at fixing it and send up a pull request that would probably be faster :)

**Comment by indieocean** [comment]
+1

**Comment by VijayanDakshinamoorthy** [comment]
Has this not been fixed yet? Did anybody find a way out? thanks.

**Comment by shahfaisal14** [comment]
Facing the same issue, using v1.0.5 Has this issue been solved.?

**Comment by yoonjesung** [comment]
+1

**Comment by juliohr** [comment]
Does anybody know a workaround? I tried to disable the tooltip when data is empty, but it didn't work.

**Comment by shahfaisal14** [comment]
yes juliohr. There was a workaround. You can throw a new error when there is no data on tooltip this will break the javascript execution and tooltip will not be visible at all. But in console there will be an error of unhandled exception everytime tooltip goes on the fake point. If you can go with the error in console then throw new error. This is what helped me.

**Comment by jlahare** [comment]
is there any update over this ?

**Comment by meghaanaa** [comment]
please suggest if any one have some solution to this problem..

**Comment by dgofman** [comment]
Create a function nodata:

```
//  nv.models.lineWithFocusChart
function nodata(noDataText) {
       svg.datum([{
            values: [[]]
        }]).call(chart);
        svg.datum([]).call(chart.noData(noDataText));
        chart.interactiveLayer.tooltip.data([]);
}
```

or

```
//  nv.models.lineChart
function nodata(noDataText) {
       svg.datum([{
            values: [[]]
        }]).call(chart);
        svg.datum([]).call(chart.noData(noDataText));
        chart.tooltip.data([]);
}
```
